extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -612,6 +643,190 @@ BeanDefiningAnnotationBuildItem quarkusMain() {
         return new BeanDefiningAnnotationBuildItem(DotName.createSimple(QuarkusMain.class.getName()), DotNames.SINGLETON);
     }
 
+    @BuildStep
+    AnnotationsTransformerBuildItem transformListAllInjectionPoints() {
+        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+
+            @Override
+            public int getPriority() {
+                return Integer.MIN_VALUE;
+            }
+
+            @Override
+            public boolean appliesTo(Kind kind) {
+                return kind == Kind.FIELD || kind == Kind.METHOD;
+            }
+
+            @Override
+            public void transform(TransformationContext ctx) {
+                if (Annotations.contains(ctx.getAnnotations(), DotNames.ALL)) {
+                    // For each injection point annotated with @All add a synthetic qualifier 
```

Comment:
Out of curiosity, why is this necessary?

Reply:
That's  a good question. Imagine we have the following injection points:

```java
   @Inject
   @All
   List<Converter> converters;

   @Inject
   @All
   @MyQualifier
   List<Converter> convertersMyQualifier;
```
We need to generate two synthetic beans here. One for `converters` to perform a dynamic lookup identical to `@Inject @Any Instance<Converter>` and one for `convertersMyQualifier` to perfom a dynamic lookup identical to `@Inject @MyQualifier Instance<Converter>`. The first one must satisfy injection points with required type `List<Converter>` and qualifier `@All`. The other one must satisfy injection points with required type `List<Converter>` and qualifiers `@All` and  `@MyQualifier`. The problem is that if we don't add a synthetic qualifier `@Identified("foo")` (where foo is a hash of required type and required qualifiers) then the second bean would also satisfy the first injection point - the required type matches and the bean has qualifier `@All`. Does it make sense?

Reply:
Aaaaaha! Yes it does, I suspected there's a good reason for it, I just couldn't see it.

I guess it's because `@All` is a regular qualifier -- if it was a special annotation, things would probably look a lot different.

Reply:
> I guess it's because `@All` is a regular qualifier -- if it was a special annotation, things would probably look a lot different.

Well, it needs to be a qualifier because it's a regular injection point. If it wasn't then `@Inject @All List<T>` would be satisfied with a any bean that has `List<T>` in its set of bean types and qualifier `@Default`.

Reply:
I'm pretty sure we could define `@All` to be a special annotation that affects how `@Inject` works. Qualifier works too, for sure.

Reply:
> I'm pretty sure we could define `@All` to be a special annotation that affects how `@Inject` works.

Actually that's probably why I originally suggested `@InjectAll` :-) But as I said, standalone `@All` is fine too.

Reply:
We could but I don't think it's a good idea to override the type-safe resolution rules. It could be even more confusing :-).